physics/noahmpdrv.F90

```diff
@@ -986,11 +1052,49 @@ subroutine noahmpdrv_run                                       &
             zvfun(i) = sqrt(tem1 * tem2)
             gdx=sqrt(garea(i))
 
+       if ( .not. do_mynnsfclay) then   !GFS sfcdiff
+
       call       stability                                                               &
         (zf(i), zvfun(i), gdx, virtual_temperature, vptemp,wind(i), z0_total, z0h_total, & 
          tvs1, con_g, thsfc_loc,                                                         &
          rb1(i), fm1(i), fh1(i), fm101(i), fh21(i), cm(i), ch(i), stress1(i), ustar1(i))
 
+       rmol1(i) = undefined  !not used in GFS sfcdif -> to satsify output
+       flhc1(i) = undefined
+       flqc1(i) = undefined
```

Comment:
@barlage If I delete these three lines, the compiling will complaint that inout variables should not be undefined. So I need to allocate and initialize them GFS_typedefs.F90.

Reply:
Have you tried to define them as optional in noahmpdrv?